internal/controller/common/numaflowtypes/pipeline.go

```diff
@@ -91,7 +91,7 @@ func GetRolloutForPipeline(ctx context.Context, c client.Client, pipeline *unstr
 	return pipelineRollout, nil
 }
 
-func GetPipelinesForRollout(ctx context.Context, c client.Client, pipelineRollout *apiv1.PipelineRollout, live bool) (*unstructured.UnstructuredList, error) {
+func GetPipelinesForRollout(ctx context.Context, c client.Client, pipelineRollout *apiv1.PipelineRollout, live bool) (unstructured.UnstructuredList, error) {
```

Comment:
I agree that we can avoid `nil` check after returning struct instead of pointer but also It can affect performance in case of larger dataset as it has to copy the complete data to return.

Reply:
Good question. This is actually the UnstructuredList struct:
```
type UnstructuredList struct {
	Object map[string]interface{}

	// Items is a list of unstructured objects.
	Items []Unstructured `json:"items"`
}
```

It's not doing a deep copy on returning, so really setting both the map and setting the slice just copies a reference to data rather than copying all of the data.